internal/provider/kubernetes/kubernetes.go

```diff
@@ -3,26 +3,42 @@ package kubernetes
 import (
 	"context"
 	"fmt"
+	"sync"
 
+	"github.com/telepresenceio/watchable"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/rest"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	gwapiv1b1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	"github.com/envoyproxy/gateway/internal/envoygateway"
 	"github.com/envoyproxy/gateway/internal/envoygateway/config"
 )
 
+// ResourceTable is a listing of all of the Kubernetes resources bing
+// watched.
+type ResourceTable struct {
```

Comment:
this should be moved under `internal/provider/provider` since this will be the common resource structure for all providers (i.e. all platform specific providers will retrieve the same resources (Gateway etc) in the same format)

Reply:
Agreed but having ResourceTable in the provider pkg causes an import cycle. I say we keep it as-is for now since kube is the only supported provider for v0.2.0.